base/linalg/symmetric.jl

```diff
@@ -107,6 +107,21 @@ end
     r
 end
 
+function setindex!(A::Symmetric, v, i::Integer, j::Integer)
+    i == j || throw(ArgumentError("Cannot set a non-diagonal index in a symmetric matrix"))
```

Comment:
It cripples the use of the wrapper though (enough that it can basically only be used for dispatch). There need to be some way to assemble into the wrapped matrix.. I'm fine with disallowing setting elements on the "wrong" side of the diagonal but imo setting on the correct side should be allowed.